### PR TITLE
Add support for the Reporting resources

### PR DESCRIPTION
**Pull request by remi-stripe**
Adding support for the new Reporting resources. Those are in OpenAPI but not yet public but since they are starting to be used and are stable we should merge them.

It's WIP for now while I get some feedback on the implementation and the shape of the resources.

cc @stripe/api-libraries 
cc @brahn-stripe

### CODE REVIEW

**Comment by remi-stripe** [comment]
We resolved all the issues internally and it's ready to be merged.

r? @brandur-stripe

File: lib/Reporting/ReportType.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * @property string $id
+ * @property string $object
+ * @property int $data_available_end
+ * @property int $data_available_start
+ * @property string $name
+ * @property int $updated
+ * @property string $version
+ * @property mixed $result
+ * @property string $status
+ * @property int $succeeded_at
```
Comment (brandur-stripe) [inline lib/Reporting/ReportType.php]: I'm not seeing `result`/`status`/`succeeded_at` in the implementation. Are you sure those are still relevant?

Comment (remi-stripe) [reply lib/Reporting/ReportType.php]: hmmm looks like a bad copy paste, fixing

Comment (remi-stripe) [reply lib/Reporting/ReportType.php]: Fixed

**Comment by remi-stripe** [comment]
@brandur-stripe PTAL (And thank you for catching this mistake)

**Comment by brandur-stripe** [comment]
LGTM.

**Comment by brandur-stripe** [comment]
Released as 6.17.0.

**Pull request merged**